packit_service/worker/github_handlers.py

```diff
@@ -522,68 +523,74 @@ def run(self) -> HandlerResults:
-#         r = BuildStatusReporter(self.project, self.event.commit_sha)
-#         if self.event.github_login not in collaborators:
-#             msg = "Only collaborators can trigger Packit-as-a-Service"
-#             r.set_status("failure", msg, PRCheckName.get_build_check())
-#             return HandlerResults(success=False, details={"msg": msg})
-#
-#         job = self.get_build_metadata_for_build()
-#
-#         dist_git_branch = (
-#             "master" if not job else job.metadata.get("dist-git-branch", "master")
-#         )
-#         try:
-#             self.api.build(dist_git_branch=dist_git_branch)
-#         except PackitException as ex:
-#             msg = (
-#                 f"There was an error while building "
-#                 f"a Fedora package `{self.package_config.downstream_package_name}`."
-#             )
-#             self.project.pr_comment(self.event.pr_id, msg)
-#             logger.error(f"error while running a build: {ex}")
-#             return HandlerResults(success=False, details={})
-#
-#         return HandlerResults(success=True, details={})
+@add_to_pr_comment_mapping
+class GitHubIssueCommentProposeUpdateHandler(PullRequestCommentHandler):
+    """ Issue handler for comment `/packit propose-update` """
+
+    name = PullRequestCommentAction.propose_update
+    event: IssueCommentEvent
+
+    def __init__(self, config: Config, event: IssueCommentEvent):
+        super().__init__(config=config, event=event)
+
+        self.config = config
+        self.event = event
+        self.project: GitProject = self.github_service.get_project(
+            repo=event.base_repo_name, namespace=event.base_repo_namespace
+        )
+        # Get the latest pull request commit
+        self.event.tag_name = self.project.get_latest_release().tag_name
+        self.package_config: PackageConfig = get_package_config_from_repo(
+            self.project, self.event.tag_name
+        )
+        self.package_config.upstream_project_url = event.https_url
+
+    def get_build_metadata_for_build(self) -> List[str]:
+        """
+        Check if there are propose-update defined
+        :return: JobConfig or Empty list
+        """
+        return [
+            job.metadata.get("dist-git-branch")
+            for job in self.package_config.jobs
+            if job.job == JobType.propose_downstream
+        ]
+
+    def run(self) -> HandlerResults:
+        self.local_project = LocalProject(
+            git_project=self.project, working_dir=self.config.command_handler_work_dir
+        )
+
+        self.api = PackitAPI(self.config, self.package_config, self.local_project)
+
+        collaborators = self.project.who_can_merge_pr()
+        r = BuildStatusReporter(self.project, self.event.tag_name)
+        if self.event.github_login not in collaborators:
+            msg = "Only collaborators can trigger Packit-as-a-Service"
+            r.set_status("failure", msg, PRCheckName.get_build_check())
```

Comment:
where this check is set?

Reply:
This is set in section, where are CI trigger, Zuul status etc.

Reply:
But isn't this handler working just with the issue?

Reply:
IT should be a comment. Definitely. Good catch